Sources/TuistAutomation/Utilities/TargetRunner.swift

```diff
@@ -0,0 +1,162 @@
+            configuration: configuration
+        )
+        let runnablePath = xcodeBuildDirectory.appending(component: target.target.productNameWithExtension)
+        guard FileHandler.shared.exists(runnablePath) else {
+            throw TargetRunnerError.runnableNotFound(path: runnablePath.pathString)
+        }
+
+        switch (target.target.platform, target.target.product) {
+        case (.macOS, .commandLineTool):
+            try runExecutable(runnablePath, arguments: arguments)
+        case let (platform, .app):
+            try runApp(
+                target: target,
+                schemeName: schemeName,
+                configuration: configuration,
+                appPath: runnablePath,
+                platform: platform,
+                minVersion: minVersion,
+                version: version,
+                deviceName: deviceName,
+                arguments: arguments
+            )
+        default:
+            throw TargetRunnerError.runningNotSupported(target: target.target)
+        }
+    }
+
+    public func assertCanRunTarget(_ target: Target) throws {
+        switch (target.platform, target.product) {
+        case (.macOS, .commandLineTool),
+             (_, .app):
+            break
+        default:
+            throw TargetRunnerError.runningNotSupported(target: target)
+        }
+    }
+
+    private func runExecutable(_ executablePath: AbsolutePath, arguments: [String]) throws {
+        logger.notice("Running executable \(executablePath.basename)", metadata: .section)
+        logger.debug("Forwarding arguments: \(arguments.joined(separator: ", "))")
+        try System.shared.runAndPrint([executablePath.pathString] + arguments)
+    }
+
+    private func runApp(
+        target: ValueGraphTarget,
+        schemeName: String,
+        configuration: String,
+        appPath: AbsolutePath,
+        platform: Platform,
+        minVersion: Version?,
+        version: Version?,
+        deviceName: String?,
+        arguments: [String]
+    ) throws {
+        let buildTarget: XcodeBuildTarget = {
+            if target.project.xcodeProjPath.extension == ".xcworkspace" {
+                return XcodeBuildTarget.workspace(target.project.xcodeProjPath)
+            } else {
+                return XcodeBuildTarget.project(target.project.xcodeProjPath)
+            }
+        }()
+        let settings = try xcodeBuildController.showBuildSettings(buildTarget, scheme: schemeName, configuration: configuration)
+            .toBlocking()
+            .single()
+        let bundleId = settings[target.target.name]?.productBundleIdentifier ?? target.target.bundleId
+        let simulator = try simulatorController
+            .findAvailableDevice(platform: platform, version: version, minVersion: minVersion, deviceName: deviceName)
+            .toBlocking()
+            .single()
+
+        logger.debug("Running app \(appPath.pathString) with arguments [\(arguments.joined(separator: ", "))]")
```

Comment:
Throwing an idea for future improvements. I think we could remember the last simulator used by the user and use that when no device/os are provided, and the remembered simulator is available.